bitcoin: broken on aarch64-darwin

the build ends up with:

```
$ nix-shell -p bitcoind
...
...
buildPhase completed in 3 minutes 50 seconds
running tests
check flags: -j8 -l8 SHELL=/nix/store/xksljrjzgx6qc6xvv2gz81vg27xf3rki-bash-5.1-p16/bin/bash LC_ALL=en_US.UTF-8 check
Making check in src
make[1]: Entering directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make[2]: Entering directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make  check-TESTS check-local
make[3]: Entering directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
Running tests: addrman_tests from test/addrman_tests.cpp
make[4]: Entering directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
Running tests: base58_tests from test/base58_tests.cpp
Running tests: base32_tests from test/base32_tests.cpp
/nix/store/xksljrjzgx6qc6xvv2gz81vg27xf3rki-bash-5.1-p16/bin/bash: line 1: 92424 Killed: 9               test/test_bitcoin --catch_system_errors=no -l test_suite -t "`cat test/base58_tests.cpp | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" -- DEBUG_LOG_OUT > test/base58_tests.cpp.log 2>&1
/nix/store/xksljrjzgx6qc6xvv2gz81vg27xf3rki-bash-5.1-p16/bin/bash: line 1: 92425 Killed: 9               test/test_bitcoin --catch_system_errors=no -l test_suite -t "`cat test/base32_tests.cpp | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" -- DEBUG_LOG_OUT > test/base32_tests.cpp.log 2>&1
make[3]: *** [Makefile:19610: test/base58_tests.cpp.test] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [Makefile:19610: test/base32_tests.cpp.test] Error 1
/nix/store/xksljrjzgx6qc6xvv2gz81vg27xf3rki-bash-5.1-p16/bin/bash: line 1: 92377 Killed: 9               test/test_bitcoin --catch_system_errors=no -l test_suite -t "`cat test/addrman_tests.cpp | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" -- DEBUG_LOG_OUT > test/addrman_tests.cpp.log 2>&1
make[3]: *** [Makefile:19610: test/addrman_tests.cpp.test] Error 1
PASS: univalue/test/unitester
PASS: univalue/test/object
PASS: univalue/test/no_nul
PASS: minisketch/test
============================================================================
Testsuite summary for Bitcoin Core 23.0.0
============================================================================
# TOTAL: 4
# PASS:  4
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[4]: Leaving directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make[3]: Leaving directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make[2]: *** [Makefile:17911: check-am] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make[1]: *** [Makefile:17569: check-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-bitcoind-23.0.drv-0/bitcoin-23.0/src'
make: *** [Makefile:817: check-recursive] Error 1
error: builder for '/nix/store/v3nxvlbki9hbmd6x29gv96mvq7mna5rs-bitcoind-23.0.drv' failed with exit code 2
```

--- a/pkgs/applications/blockchains/bitcoin/default.nix
+++ b/pkgs/applications/blockchains/bitcoin/default.nix
@@ -76,7 +76,7 @@ stdenv.mkDerivation rec {
 
   # fix "Killed: 9  test/test_bitcoin"
   # https://github.com/NixOS/nixpkgs/issues/179474
-  hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "stackprotector" ];
+  hardeningDisable = lib.optionals (stdenv.isAarch64 && stdenv.isDarwin) [ "fortify" "stackprotector" ];
 
   checkInputs = [ python3 ];
 

